libraries/Microsoft.Bot.Builder/StreamingRequestHandler.cs

```diff
@@ -0,0 +1,220 @@
+            {
+                return await ProcessStreamingRequestAsync(request, response, logger, cancellationToken).ConfigureAwait(false);
+            }
+
+            response.StatusCode = (int)HttpStatusCode.NotFound;
+            logger.LogInformation($"Unknown verb and path: {request.Verb} {request.Path}");
+
+            return response;
+        }
+
+        /// <summary>
+        /// Build and return versioning information used for telemetry, including:
+        /// The Schema version is 3.1, put into the Microsoft-BotFramework header,
+        /// Protocol Extension Info,
+        /// The Client SDK Version
+        ///  https://github.com/Microsoft/botbuilder-dotnet/blob/d342cd66d159a023ac435aec0fdf791f93118f5f/doc/UserAgents.md,
+        /// Additional Info.
+        /// https://github.com/Microsoft/botbuilder-dotnet/blob/d342cd66d159a023ac435aec0fdf791f93118f5f/doc/UserAgents.md.
+        /// </summary>
+        /// <returns>A string containing versioning information.</returns>
+        private static string GetUserAgent() =>
+            string.Format(
+                "Microsoft-BotFramework/3.1 Streaming-Extensions/1.0 BotBuilder/{0} ({1}; {2}; {3})",
+                ConnectorClient.GetClientVersion(new ConnectorClient(new Uri("http://localhost"))),
+                ConnectorClient.GetASPNetVersion(),
+                ConnectorClient.GetOsVersion(),
+                ConnectorClient.GetArchitecture());
+
+        /// <summary>
+        /// Performs the actual processing of a request, handing it off to the adapter and returning the response.
+        /// </summary>
+        /// <param name="request">A ReceiveRequest from the connected channel.</param>
+        /// <param name="response">The response to update and return, ultimately sent to client.</param>
+        /// <param name="logger">Optional logger.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        /// <returns>The response ready to send to the client.</returns>
+        private async Task<StreamingResponse> ProcessStreamingRequestAsync(ReceiveRequest request, StreamingResponse response, ILogger<RequestHandler> logger, CancellationToken cancellationToken)
+        {
+            var body = request.ReadBodyAsString();
+            if (string.IsNullOrEmpty(body) || request.Streams.Count == 0)
+            {
+                response.StatusCode = (int)HttpStatusCode.BadRequest;
+                logger.LogInformation("Request missing body and/or streams.");
+
+                return response;
+            }
+
+            try
+            {
+                var adapter = new BotFrameworkStreamingExtensionsAdapter(_transportServer, _middlewareSet, logger);
+
+                // Check for dependency injected bot first, then check for bot passed as parameter, throw if neither is found.
+                var bot = _services?.GetService<IBot>() ?? this._bot ?? throw new Exception("Unable to find bot when processing request.");
```

Comment:
Too complex. Please simplify.

Reply:
I can break it up into multiple lines and statements with comments at each step, but the process will still be the same. Would that be helpful at all? 

This method acts as the "controller" for incoming in-proc requests, the service collection check is to be able to respect the transient/singleton type registration that would have been dependency injected into the controller. From what I've seen that's usually the case with NetCore bots. With WebApi bots, however, it looks like the bot is usually passed in as a parameter and treated as a singleton, so our request handler has to accommodate both. 

The only real simplification I can think of is setting a flag based on what ctor is used and going by it to know which bot to look for, instead of checking both. I'm not sure that would be less complex or just adding another var into the mix. 

Let me know if you'd prefer either of those approaches to the current, and I'll change the code accordingly!